dask/blockwise.py

```diff
@@ -119,15 +122,33 @@ def blockwise(
         kwargs2 = (dict, list(map(list, kwargs.items())))
         subgraph = {output: (apply, func, _keys, kwargs2)}
 
+    into_kwargs = {}
+    if func == operator.getitem and getitem_key is not None:
+        into = BlockwiseGetitem
+        into_kwargs["getitem_key"] = getitem_key
```

Comment:
Alternatively we could also be explicit about using a `BlockwiseGetitem` Mapping type when constructing graphs.  I would not be surprised if in the future Dask graph construction looks a lot more like selecting from a variety of types of mappings rather than constructing dicts by hand.  I don't know though, and I'd welcome conversation on this topic if you have experience here.

Reply:
Yes I think I like that suggestion better.

Reply:
Ahh, this might be a bit hard though. The call stack is roughly 

```
DataFrame.__getitem__ ->
  partitionwise_graph ->
    blockwise ->
      Blockwise (or a BlockwiseGetitem, BlockwiseFilter, etc.)
```

So it's a bit hard for `DataFrame.__getitem__` to specify the mapping type. We would presumably need something like a `partitionwise_getitem_graph`, and `partitonwise_getitem`, which doesn't feel much cleaner.